django/contrib/postgres/indexes.py

```diff
@@ -57,3 +57,37 @@ def get_sql_create_template_values(self, model, schema_editor, using):
 
     def create_sql(self, model, schema_editor):
         return super().create_sql(model, schema_editor, using=' USING gin')
+
+
+class GistIndex(Index):
+    suffix = 'gist'
+    # Allow an index name longer than 30 characters since the suffix is 4
```

Comment:
I'd move this to a mixin to share with BrinIndex or drop the repetitive comment.